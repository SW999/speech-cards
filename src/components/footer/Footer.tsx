import * as React from 'react';
import { FunctionComponent } from 'react';
import { version } from '../../../package.json';
import github from '../../img/github.svg';

export const Footer: FunctionComponent = () => (
  <div className="footer">
    <a
      href="https://github.com/SW999/speech-cards"
      rel="nofollow"
      title="Github page"
    >
      <img src={github} alt="github" width="24" height="24" />
    </a>
    &nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;Â© SW999
    2020&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;
    {`v. ${version}`}
    &nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;
    <a
      href="mailto:swarog2002@mail.ru?subject=Speech%20cards"
      target="_blank"
      rel="noopener noreferrer"
      title="Email me"
    >
      <span role="img" aria-label="email me">
        ðŸ“§
      </span>
    </a>
  </div>
);
