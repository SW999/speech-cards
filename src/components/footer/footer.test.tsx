import * as React from 'react';
import { render } from '@testing-library/react';
import { Footer } from './Footer';

jest.mock(
  '../../../package.json',
  () => ({
    version: '1.0.0',
  }),
  { virtual: true }
);

describe('Footer component', () => {
  it('Footer should take a snapshot', () => {
    const { asFragment } = render(<Footer />);

    expect(asFragment()).toMatchSnapshot('Footer');
  });
});
