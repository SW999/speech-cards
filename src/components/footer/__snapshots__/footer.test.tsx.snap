// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Footer should take a snapshot: Footer 1`] = `
<DocumentFragment>
  <div
    class="footer"
    role="contentinfo"
  >
    <a
      href="https://github.com/SW999/speech-cards"
      rel="nofollow"
      title="Github page"
    >
      <img
        alt="github"
        height="24"
        src="[object Object]"
        width="24"
      />
    </a>
    Â Â Â |Â Â Â Â© SW999 2020Â Â Â |Â Â Â v. 1.0.0Â Â Â |Â Â Â 
    <a
      href="mailto:swarog2002@mail.ru?subject=Speech%20cards"
      rel="noopener noreferrer"
      target="_blank"
      title="Email me"
    >
      <span
        aria-label="email me"
        role="img"
      >
        ðŸ“§
      </span>
    </a>
  </div>
</DocumentFragment>
`;
